ohshie.HabitTracker/DbOperations.cs

```diff
@@ -0,0 +1,263 @@
+
+            SqliteDataReader reader = tableCommand.ExecuteReader();
+
+            if (reader.HasRows)
+            {
+                while (reader.Read())
+                {
+                    habitData.Add(
+                        new TrackedHabits()
+                        {
+                            Id = reader.GetInt32(0),
+                            Date = reader.GetString(1),
+                            Quantity = reader.GetInt32(2),
+                            AmountType = GetHabitQuantifier(habitName)
+                        });
+                }
+            }
+            else
+            {
+                Console.WriteLine("Database empty!\n" +
+                                  "Press any key to go back");
+                Console.ReadKey(true);
+                return -1;
+            }
+            
+            connection.Close();
+
+            Console.WriteLine("######################");
+            foreach (var entry in habitData)
+            {
+                Console.WriteLine($"{entry.Id}. On {entry.Date}. You've read {entry.Quantity} {entry.AmountType}");
+            }
+            Console.WriteLine("######################");
+        }
+
+        return 1;
+    }
+
+    public void DeleteEntry(string dbName)
+    {
+        Console.Clear();
+        
+        PrintTable(dbName);
+        
+        using (var connection = new SqliteConnection(Dbconnection))
+        {
+            connection.Open();
+            
+            var tableCommand = connection.CreateCommand();
+            int invalidRow = 0;
+            int entryId = 0;
+            
+            while (invalidRow == 0)
+            {
+                entryId = Program.GetNumberFromUser("Id of entry you want to delete");
+                if (entryId == -1) return;
+
+                tableCommand.CommandText = $"delete from {dbName} where id = '{entryId}'";
+
+                invalidRow = tableCommand.ExecuteNonQuery();
+                if (invalidRow == 0) Console.WriteLine("Looks like there you've chosen entry that does not exist, or there are no entries left");
+            }
+            
+            connection.Close();
+            
+            Console.WriteLine($"Entry with Id {entryId} was deleted");
+        }
+    }
+
+    public void UpdateEntry(string dbName)
+    {
+        Console.Clear();
+        
+        using (var connection = new SqliteConnection(Dbconnection))
+        {
+            connection.Open();
+            var tableCommand = connection.CreateCommand();
+            int entryId = 0;
+            int checkEntryExist = 0;
+            
+            while (checkEntryExist == 0)
+            {
+                PrintTable(dbName);
+                entryId = Program.GetNumberFromUser("Id of entry you want to edit");
+                if (entryId == -1) return;
+                
+                tableCommand.CommandText = $"select exists(select 1 from {dbName} where Id = {entryId})";
```

Comment:
For better readability, please change all SQL keywords in your code to capital letters. In this line, it would become SELECT EXISTS(SELECT 1 FROM {dbName} WHERE Id = {entryId})".

I'm going to add that to the code conventions. 😊😊